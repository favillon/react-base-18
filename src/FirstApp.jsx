import PropTypes from 'prop-types'

const nametUser = 'Andres';
const obj = {
  msg  : 'Hola',
  msg2 : 'Mundo'
}
const myName = (name = nametUser) =>{
  return `Name is ${name}`
}

export const FirstApp = ({title, number, subtitle}) => {
  return (
    <>
      <h1>{title}</h1>
      <h2 data-testid="test-subtitle">{subtitle}</h2>
      <h3>{ nametUser } : {number}</h3>
      <h3>{ myName() }</h3>
      <h3>{ myName('Paco') }</h3>
      <code>{ JSON.stringify(obj) }</code>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  number : PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  subtitle : 'Sin Subtitulo'
}