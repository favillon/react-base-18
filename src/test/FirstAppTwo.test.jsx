import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Test in <FirstApp>', () => {
    const title ='Soy el titulo';
    test('should match with snapshot', () => {
        const { container } = render(<FirstApp title={title} number={1} />);

        expect(container).toMatchSnapshot()
    })
    test('should view title in h1', () => {
        render(<FirstApp title={title} number={1} />);

        //screen.debug()
        expect(screen.getByText(title)).toBeTruthy();
        const h1 = screen.getByRole('heading', {level: 1});
        expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);
    })
    test('should view subtitle in h2', () => {
        render(<FirstApp title={title} number={1} />);

        expect(screen.getByTestId('test-subtitle')).toBeTruthy();
        expect(screen.getByTestId('test-subtitle').innerHTML).toBe('Sin Subtitulo');
    })
    test('should view subtitle in h3 for three', () => {
        
        render(<FirstApp title={title} number={1} />);
        const h3 = screen.queryAllByRole('heading', {level: 3});
        expect(h3.length).toBe(3);
    })
    test('should view nuumber in h3', () => {
        /*const numberProps = 1
        render(<FirstApp title={title} number={numberProps} />);
        const h3 = screen.queryAllByRole('heading', {level: 3});
        console.log(h3);
        expect(h3).toContain(numberProps.toString());*/
    })
})