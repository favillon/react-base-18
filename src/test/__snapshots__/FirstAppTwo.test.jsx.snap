// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp> should match with snapshot 1`] = `
<div>
  <h1>
    Soy el titulo
  </h1>
  <h2
    data-testid="test-subtitle"
  >
    Sin Subtitulo
  </h2>
  <h3>
    Andres
     : 
    1
  </h3>
  <h3>
    Name is Andres
  </h3>
  <h3>
    Name is Paco
  </h3>
  <code>
    {"msg":"Hola","msg2":"Mundo"}
  </code>
</div>
`;
